src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -195,11 +200,19 @@ trait FastSync {
       case SyncRequestHandler.Done =>
         context unwatch sender()
         assignedHandlers -= sender()
+        requestedMptNodes = requestedMptNodes - sender()
+        requestedNonMptNodes = requestedNonMptNodes - sender()
+        requestedBlockBodies = requestedBlockBodies - sender()
+        requestedReceipts = requestedReceipts - sender()
         processSyncing()
 
       case Terminated(ref) if assignedHandlers.contains(ref) =>
         context unwatch ref
         assignedHandlers -= ref
+        requestedMptNodes = requestedMptNodes - ref
+        requestedNonMptNodes = requestedNonMptNodes - ref
+        requestedBlockBodies = requestedBlockBodies - ref
+        requestedReceipts = requestedReceipts - ref
```

Comment:
Terminated message will happen after requested entities will be added back to queue?